src/devices/machine/cs8900a.cpp

```diff
@@ -0,0 +1,1522 @@
+CS8900_PP_ADDR_TXLENGTH        0x0146 * # -W - 4.5., p. 70 - 5.7., p. 98 *
+CS8900_PP_ADDR_MAC_ADDR        0x0158 * # RW - 4.6., p. 71 - 5.3., p. 86 *
+                            0x015a
+                            0x015c
+*/
+
+/* ------------------------------------------------------------------------- */
+/* TX/RX buffer handling */
+
+void cs8900a_device::cs8900_write_tx_buffer(u8 value,int odd_address)
+{
+    /* write tx data only if valid buffer is ready */
+    if(tx_state != CS8900_TX_READ_BUSST) {
+#ifdef CS8900_DEBUG_WARN_RXTX
+        log_message(cs8900_log, "WARNING! Ignoring TX Write without correct Transmit Condition! (odd=%d,value=%02x)",
+                    odd_address,value);
+#endif
+        /* ensure correct tx state (needed if transmit < 4 was started) */
+        cs8900_set_tx_status(0,0);
+    } else {
+#ifdef CS8900_DEBUG_RXTX_STATE
+        if(tx_count==0) {
+            log_message(cs8900_log,"TX: write frame (length=%04x)",tx_length);
+        }
+#endif
+
+        /* always write LH, LH... to tx buffer */
+        u16 addr = tx_buffer;
+        if(odd_address) {
+            addr++;
+            tx_buffer += 2;
+        }
+        tx_count++;            
+        SET_PP_8(addr, value);
+        
+#ifdef CS8900_DEBUG_RXTX_DATA
+        log_message(cs8900_log, "TX: %04x/%04x: %02x (buffer=%04x,odd=%d)",
+                    tx_count,tx_length,value,addr,odd_address);
+#endif                
+
+        /* full frame transmitted? */
+        if(tx_count==tx_length) {
+#ifdef CS8900_DEBUG_FRAMES
+            log_message(cs8900_log, "cs8900_arch_transmit() called with:                 "
+                "length=%4u and buffer %s", tx_length,
+                debug_outbuffer(tx_length, &cs8900_packetpage[CS8900_PP_ADDR_TX_FRAMELOC])
+                );
+#endif
+
+            if(!tx_enabled) {
+#ifdef CS8900_DEBUG_WARN_RXTX
+                log_message(cs8900_log,"WARNING! Can't transmit frame (Transmitter is not enabled)!");
+#endif
+            } else {
+                /* send frame */
+#ifdef CS8900_DEBUG
+                printf("SENDING from buf %p len %d\n", &cs8900_packetpage[CS8900_PP_ADDR_TX_FRAMELOC], tx_length);
+#endif
+                send(&cs8900_packetpage[CS8900_PP_ADDR_TX_FRAMELOC],tx_length);
+            }
+
+            /* reset transmitter state */
+            tx_state = CS8900_TX_IDLE;
+    
+#ifdef CS8900_DEBUG_RXTX_STATE
+            log_message(cs8900_log, "TX: sent  frame (length=%04x)",tx_length);
+#endif
+
+            /* reset tx status */
+            cs8900_set_tx_status(0,0);
+        }
+    }
+}
+
+u8 cs8900a_device::cs8900_read_rx_buffer(int odd_address)
+{
+    if(rx_state != CS8900_RX_GOT_FRAME) {
+#ifdef CS8900_DEBUG_WARN_RXTX
+        log_message(cs8900_log, "WARNING! RX Read without frame available! (odd=%d)",
+                    odd_address);
+#endif
+        /* always reads zero on HW */
+        return 0;
+    } else {
```

Comment:
Please be consistent with your style – most of your `else` are on the line following the `}` of the preceding compound statement, but this one is on the same line.

Reply:
Roger that.  I found a Google+Allman uncrustify and will be applying it after the functional fixes.